xpublish_edr/formats/to_geotiff.py

```diff
@@ -0,0 +1,63 @@
+"""
+Generate GeoTIFF responses for an xarray dataset for EDR queries
+"""
+
+import xarray as xr
+from fastapi import Response
+
+
+def to_geotiff(ds: xr.Dataset) -> Response:
+    """Return a GeoTIFF response from an xarray dataset"""
+    import io
+
+    import numpy as np
+    import rasterio
+    from rasterio.transform import from_origin
+
+    # Get the first data variable
+    var_name = list(ds.data_vars)[0]
+    data = ds[var_name].values
+
+    # Handle different dimensions
+    if len(data.shape) == 2:
+        # 2D data (x,y)
+        data = np.expand_dims(data, axis=0)
+    elif len(data.shape) > 3:
+        # More than 3D, take first slice
+        data = data[0]
+
+    # Get coordinates
+    x_coords = ds[ds[var_name].dims[-1]].values
+    y_coords = ds[ds[var_name].dims[-2]].values
+
+    # Calculate pixel size
+    x_res = (x_coords[-1] - x_coords[0]) / (len(x_coords) - 1)
+    y_res = (y_coords[-1] - y_coords[0]) / (len(y_coords) - 1)
+
+    # Create transform
+    transform = from_origin(x_coords[0], y_coords[-1], x_res, y_res)
+
+    # Create in-memory GeoTIFF
+    memfile = io.BytesIO()
```

Comment:
Would there be any reason to use a fastapi [streaming response](https://fastapi.tiangolo.com/advanced/custom-response/#using-streamingresponse-with-file-like-objects) rather than putting the whole file into memory? Would this allow us to deal with bigger files?

Reply:
Im not sure how rasterio deals with streaming yet, have to figure that out